### PR TITLE
 Fix Kubernetes Destination Plugin 

### PR DESCRIPTION
**Pull request by wfhartford**
The Kubernetes plugin was broken. There were two major issues:
* The server certificate was entered in a string input making it impossible (as far as I know) to enter a valid PEM certificate.
* The base64 encoding calls were passing strings where bytes were expected.

The fix to the first issue depends on #2218 and a change in the options structure. I've also included some improved input validation and logging.

### CODE REVIEW

**Comment by wfhartford** [comment]
This should address  #1790. As I commented there, I'll soon submit another PR with a couple minor improvements.

Fair Warning: I'm not much of a python programmer. I've probably done things that aren't very pythonic. Feel free to let me know.

File: lemur/plugins/lemur_kubernetes/plugin.py [hunk 1]
```diff
@@ -26,21 +28,32 @@ def ensure_resource(k8s_api, k8s_base_uri, namespace, kind, name, data):
...
 
     create_resp = k8s_api.post(url, json=data)
+    current_app.logger.debug("K8S POST response: %s", create_resp)
 
     if 200 <= create_resp.status_code <= 299:
         return None
 
-    elif create_resp.json()['reason'] != 'AlreadyExists':
-        return create_resp.content
+    else:
```
Comment (castrapel) [inline lemur/plugins/lemur_kubernetes/plugin.py]: This whole blurb  from L39-L45 is a bit wordy. We can make it safer by just changing the original lines to the following:

`elif create_resp.json().get('reason', '') != 'AlreadyExists':
        return create_resp.content`

File: lemur/plugins/lemur_kubernetes/plugin.py [hunk 2]
```diff
@@ -26,21 +28,32 @@ def ensure_resource(k8s_api, k8s_base_uri, namespace, kind, name, data):
...
 
-    update_resp = k8s_api.put(_resolve_uri(k8s_base_uri, namespace, kind, name), json=data)
+    update_resp = k8s_api.put(url, json=data)
+    current_app.logger.debug("K8S PUT response: %s", update_resp)
 
     if not 200 <= update_resp.status_code <= 299:
         return update_resp.content
 
-    return
+    return None
```
Comment (castrapel) [inline lemur/plugins/lemur_kubernetes/plugin.py]: `return` and `return None` are essentially the same thing. It's more Pythonic to just use return in most cases. even without any `return`, None will be returned, but explicitly having just `return` is fine.

File: lemur/plugins/lemur_kubernetes/plugin.py [hunk 3]
```diff
@@ -106,33 +125,38 @@ def __init__(self, *args, **kwargs):
...
+                },
+                'data': {
+                    'combined.pem': base64encode('%s\n%s' % (body, private_key)),
+                    'ca.crt': base64encode(cert_chain),
+                    'service.key': base64encode(private_key),
+                    'service.crt': base64encode(body),
+                }
+            })
+        except Exception as e:
+            current_app.logger.exception("Exception in upload")
```
Comment (castrapel) [inline lemur/plugins/lemur_kubernetes/plugin.py]: if you add a parameter  here, the log can contain traceback info specifying specific details about the exception

current_app.logger.exception("Exception in upload: {}".format(e), exc_info=True)

File: lemur/plugins/lemur_kubernetes/plugin.py [hunk 4]
```diff
@@ -106,33 +125,38 @@ def __init__(self, *args, **kwargs):
...
+            raise e
```
Comment (castrapel) [inline lemur/plugins/lemur_kubernetes/plugin.py]: You can just say `raise` instead of `raise e`

File: lemur/plugins/lemur_kubernetes/plugin.py (hunk 1 above)
Comment (wfhartford) [reply lemur/plugins/lemur_kubernetes/plugin.py]: Excellent, I knew there would be a better way.

File: lemur/plugins/lemur_kubernetes/plugin.py (hunk 2 above)
Comment (wfhartford) [reply lemur/plugins/lemur_kubernetes/plugin.py]: Thanks

File: lemur/plugins/lemur_kubernetes/plugin.py (hunk 3 above)
Comment (wfhartford) [reply lemur/plugins/lemur_kubernetes/plugin.py]: Cool, thanks

File: lemur/plugins/lemur_kubernetes/plugin.py (hunk 4 above)
Comment (wfhartford) [reply lemur/plugins/lemur_kubernetes/plugin.py]: Excelent

**Comment by castrapel** [comment]
LGTM! Thanks for the help here.
